EngineLayer/PeptideSpectralMatch.cs

```diff
@@ -414,11 +437,16 @@ private static void AddPeptideSequenceData(Dictionary<string, string> s, Peptide
         {
             bool pepWithModsIsNull = psm == null || psm.BestMatchingPeptides == null || !psm.BestMatchingPeptides.Any();
 
-            var pepsWithMods = pepWithModsIsNull ? null : psm.BestMatchingPeptides.Select(p => p.Peptide).ToList();
+            List<PeptideWithSetModifications> pepsWithMods = pepWithModsIsNull ? null : psm.BestMatchingPeptides.Select(p => p.Peptide).ToList();
+
+            IEnumerable<string> baseSequences = pepWithModsIsNull ? null : pepsWithMods.Select(b => b.BaseSequence);
+            IEnumerable<string> fullSequences = pepWithModsIsNull ? null : pepsWithMods.Select(b => b.FullSequence);
+            IEnumerable<string> essentialSequences = pepWithModsIsNull ? null : pepsWithMods.Select(b => b.EssentialSequence(ModsToWritePruned));
+            var checksForDistinctNames = new List<IEnumerable<string>> { baseSequences, fullSequences, essentialSequences };
 
-            s["Base Sequence"] = pepWithModsIsNull ? " " : Resolve(pepsWithMods.Select(b => b.BaseSequence)).ResolvedString;
-            s["Full Sequence"] = pepWithModsIsNull ? " " : Resolve(pepsWithMods.Select(b => b.FullSequence)).ResolvedString;
-            s["Essential Sequence"] = pepWithModsIsNull ? " " : Resolve(pepsWithMods.Select(b => b.EssentialSequence(ModsToWritePruned))).ResolvedString;
+            s["Base Sequence"] = pepWithModsIsNull ? " " : Resolve(baseSequences).ResolvedString;
+            s["Full Sequence"] = pepWithModsIsNull ? " " : Resolve(fullSequences).ResolvedString;
+            s["Essential Sequence"] = pepWithModsIsNull ? " " : Resolve(essentialSequences).ResolvedString;
```

Comment:
instead of doing all this, check to see if fullSequence is ambiguous `(psm.FullSequence == null)` if it's ambiguous). if it's ambiguous, write out all protein accessions. else, write distinct protein accessions/names/any other protein info